src/coreclr/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -103,13 +104,19 @@ internal static unsafe IntPtr ConvertToNative(int flags, string strManaged, IntP
                     // wasting memory on systems with multibyte character sets where the buffer we end up with is often much
                     // smaller than the upper bound for the given managed string.
 
-                    byte[] bytes = AnsiCharMarshaler.DoAnsiConversion(strManaged,
-                        fBestFit: 0 != (flags & 0xFF), fThrowOnUnmappableChar: 0 != (flags >> 8), out nb);
+                    byte[] bytes = ArrayPool<byte>.Shared.Rent(checked((strManaged.Length + 1) * Marshal.SystemMaxDBCSCharSize));
```

Comment:
All of the marshallers in this file are quite legacy. There are many other opportunities to update.
I'd suggest to not touch it.